receiver/redisreceiver/redis_runnable.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redisreceiver
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/obsreport"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver/interval"
+)
+
+var _ interval.Runnable = (*redisRunnable)(nil)
+
+// Runs intermittently, fetching info from Redis, creating metrics/datapoints,
+// and feeding them to a metricsConsumer.
+type redisRunnable struct {
+	ctx             context.Context
+	metricsConsumer consumer.MetricsConsumerOld
+	redisSvc        *redisSvc
+	redisMetrics    []*redisMetric
+	logger          *zap.Logger
+}
+
+func newRedisRunnable(
+	ctx context.Context,
+	client client,
+	metricsConsumer consumer.MetricsConsumerOld,
+	logger *zap.Logger,
+) *redisRunnable {
+	return &redisRunnable{
+		ctx:             ctx,
+		redisSvc:        newRedisSvc(client),
+		metricsConsumer: metricsConsumer,
+		logger:          logger,
+	}
+}
+
+// Builds a data structure of all of the keys, types, converters and such
+// to later extract data from Redis.
+func (r *redisRunnable) Setup() error {
+	r.redisMetrics = getDefaultRedisMetrics()
+	return nil
+}
+
+// Runs intermittently, querying Redis and building Metrics to send to the
```

Comment:
>Runs intermittently

This is a bit confusing. This function performs a single run, single cycle of collection from Redis, right? When I read the comment I get the impression that Run() function internally performs multiple run cycles.

Perhaps reword a bit, like this:
>// Run is called periodically, ...

Reply:
Better, fixed.